activerecord/test/cases/adapters/postgresql/transaction_nested_test.rb

```diff
@@ -68,15 +77,56 @@ class Sample < ActiveRecord::Base
       end
     end
 
+    test "SerializationFailure inside nested SavepointTransaction is recoverable" do
+      start_right = Concurrent::Event.new
+      commit_left = Concurrent::Event.new
+      finish_right = Concurrent::Event.new
+      Sample.create value: 1
+
+      thread = Thread.new do
+        with_warning_suppression do
+          Sample.transaction(isolation: :serializable, requires_new: false) do
+            Sample.update_all value: 2
+            start_right.set
+            commit_left.wait(1)
+          end
+          finish_right.set
+        end
+      end
+
+      begin
+        with_warning_suppression do
+          start_right.wait
+          Sample.transaction(isolation: :serializable, requires_new: false) do
+            assert_raises(ActiveRecord::SerializationFailure) do
+              Sample.transaction(requires_new: true) do
+                Sample.create value: 3
+                commit_left.set
+                finish_right.wait(2)
+                Sample.update_all value: 4
+              end
+            end
+            Bit.create value: 1
```

Comment:
Hey folks 👋 
Sorry for bumping an old PR, but recently I've been working around some transaction logic and realized that some tests have slightly misleading setup, including this one.
Specifically, the test implies that it uses a `SavepointTransaction` when in reality it's not because the nested transaction is going to be an instance of `RestartParentTransaction` as long as the parent transaction has no queries.
You may want to see more details here - https://github.com/rails/rails/pull/44686

I was able to fix most of the tests by adding a dummy query to force Rails to open an actual savepoint transaction without breaking existing tests expect this one.
You can find a changed version of the test in the PR mentioned below, but when I make this test to use a savepoint transaction, it starts to fail on `Bit.create value: 1` with the following exception (also available on the PR's CI):
```ruby


ActiveRecord::SerializationFailure: PG::TRSerializationFailure: ERROR:  could not serialize access due to read/write dependencies among transactions
--
  | DETAIL:  Reason code: Canceled on identification as a pivot, during conflict in checking.
  | HINT:  The transaction might succeed if retried.
```

I'm going to dig into this, but just wanted to share this feedback just in case you may see the issue off the top of your head. Cause there is still a chance that my setup is wrong but it also plausible that assumptions made about postgresql behaviour were wrong. Don't want to make any premature assumptions yet!

Thanks!

Reply:
Just to close the loop here. I can confirm that the original test was ok and used a savepoint transaction, so this PR is not involved in the issue I'm looking into.
I can confirm that the issue is reproducible on https://github.com/rails/rails/pull/44526 but not going to say that this is the cause yet.
Just wanted to mention that this PR is unrelated, sorry for bothering! 
And thanks for the pointer ❤️